crates/polars-lazy/src/frame/mod.rs

```diff
@@ -741,7 +742,30 @@ impl LazyFrame {
                     payload: SinkType::Memory,
                 });
 
-                return polars_stream::run_query(lp_top, ir_arena, expr_arena);
+                polars_stream::run_query(lp_top, ir_arena, expr_arena)
+            };
+            if self.opt_state.new_streaming {
+                return run_on_new_streaming_engine();
+            }
+
+            if std::env::var("POLARS_AUTO_NEW_STREAMING")
+                .as_deref()
+                .unwrap_or("")
+                == "1"
+            {
+                match std::panic::catch_unwind(run_on_new_streaming_engine) {
+                    Ok(r) => return r,
+                    Err(e) => {
+                        // Fallback to normal engine if error is due to not being implemented,
+                        // otherwise propagate error.
+                        if e.downcast_ref::<&str>() != Some(&"not yet implemented") {
+                            if polars_core::config::verbose() {
+                                eprintln!("caught unimplemented error in new streaming engine, falling back to normal engine");
+                            }
+                            std::panic::resume_unwind(e);
+                        }
+                    },
+                }
```

Comment:
Can you insert something like this here?


Then we don't go into the optimizer twice.
```rust
            // let mut physical_plan =
            //     create_physical_plan(lp_top, &mut alp_plan.lp_arena, &alp_plan.expr_arena)?;
            // let mut state = ExecutionState::new();
            // return physical_plan.execute(&mut state);
```